### DISCUSSION TITLE
Project publishing issues and native component documentation issues

### DISCUSSION THREAD

**Discussion by 527395632** [microsoft/react-native-windows#10008] [Q&A] [answered]
The problem I encountered is that in the development process, press F5, the program can run normally, the interface can be seen, and the function is normal. However, when I try to publish, "XXX. Appxbundle" can be started normally after installation, but the error "0x80070002 系统找不到指定的文件" is reported during operation. My first feeling should be that I didn't package the reactnative project code and resource files together. However, I consulted a lot of materials, but I didn't find the relevant user manual. I hope I can find the answer here. Here, I first thank you for your help

My operation steps are as follows：
    Development stage：
        1：Run command line “yarn start”
![image](https://user-images.githubusercontent.com/20411721/170427479-266b2c66-25f1-4ab0-9326-54bcd84007c1.png)

        2: Write the code in "app. TSX"
![image](https://user-images.githubusercontent.com/20411721/170427697-30c10ab7-4ba8-4a9a-ac78-31397e52c468.png)

        3：Press F5 in Visual Studio 2019
![image](https://user-images.githubusercontent.com/20411721/170428196-e7ff7f6a-bb4b-4c07-8f0d-72e3fc4a77dd.png)

    Publish stage:
        1: Conditional compilation symbol “BUNDLE”
![image](https://user-images.githubusercontent.com/20411721/170429736-586db549-12ea-4e37-b4b9-afa71366e849.png)

        2：Certificate created
![image](https://user-images.githubusercontent.com/20411721/170430057-39c38a3c-1c4f-450c-a160-29146838947e.png)

        3：Publish Project
![image](https://user-images.githubusercontent.com/20411721/170430191-ff65afd3-b552-4fb1-8a90-8def91ac6597.png)

        4: Get installation package And Certificate
![image](https://user-images.githubusercontent.com/20411721/170430338-b8cd525a-00ec-4e71-b317-cd7c4b50ad61.png)

        5: Operation results
![image](https://user-images.githubusercontent.com/20411721/170430413-63b214c9-4bed-4a74-b06d-665b733a1096.png)

**Comment by jonthysell** [comment, marked as answer]
There are two kinds of bundles - the appxbundle is the packaged windows app to be installed. The JS bundle is all of your JS code pre-bundled and included.

If you build Debug, you still need to run Metro to serve your up your JS bundle.

If you build Release, the JS bundle will be pre-included in the appxbundle and you won't need Metro. It will be a "self-contained" app package.

**Reply by 527395632** [reply]
I see. Thank you very much for your help. Thank you very much!!!

In addition, I also want to ask about the path of the "react native windows" document“ https://microsoft.github.io/react-native-windows/ ". all components owned by" react native windows "are not described in this document. However, it is stated in Github that it is necessary to read the documents on the official website of" react native ". I saw the introduction of components on the official website of" react native ", but some components are specific to specific platforms. I would like to know whether there are components or APIs for" react native windows "that are not mentioned in the official website of" react native ". Where are these documents Ke

**Reply by yuya-tani** [reply]
Excuse me.

I think it's the same phenomenon,
Publish the app → "Create app package"
When I create an application, isn't the JS bundle included as well? Looks like.

I have selected "Always do" in "Generate Application Bundle".

![2022-06-29 170145](https://user-images.githubusercontent.com/22019718/176384547-1cbd4d20-990e-45a8-93d1-c1090a587d83.png)
d
Is it different from this?

I would be grateful if you could tell me how it was resolved in the end.
Best regar

**Reply by 527395632** [reply]
Sorry, I don't quite understand what you mean. Can you describe your problem in Chinese?
